netboot/netboot.go

```diff
@@ -95,3 +127,28 @@ func ConversationToNetconf(conversation []dhcpv6.DHCPv6) (*NetConf, string, erro
 	}
 	return netconf, bootfile, nil
 }
+
+// ConversationToNetconfv4 extracts network configuration and boot file URL from a
+// DHCPv4 4-way conversation and returns them, or an error if any.
+func ConversationToNetconfv4(conversation []*dhcpv4.DHCPv4) (*NetConf, string, error) {
+	var reply *dhcpv4.DHCPv4
+	var bootFileUrl string
+	for _, m := range conversation {
+		// look for a BootReply packet of type Offer containing the bootfile URL.
+		// Normally both packets with Message Type OFFER or ACK do contain
+		// the bootfile URL.
+		if m.Opcode() == dhcpv4.OpcodeBootReply && *m.MessageType() == dhcpv4.MessageTypeOffer {
+			bootFileUrl = m.BootFileNameToString()
+			reply = m
+			break
+		}
+	}
+	if reply == nil {
+		return nil, "", errors.New("no OFFER with valid bootfile URL received")
```

Comment:
what about mentioning ACK too?

```
errors.New("no OFFER or ACK with valid bootfile URL received")
```

Reply:
I'll then need to amend the logic which looks for bootfileurl, as now I am looking only into OFFERs:
```
m.Opcode() == dhcpv4.OpcodeBootReply && *m.MessageType() == dhcpv4.MessageTypeOffer 
```

Reply:
ignore this, I misread the code